src/gluonts/nursery/streaming_models/masking.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Any, Callable, Tuple
+
+import numpy as np
+import pandas as pd
+
+from gluonts.core.component import validated
+from gluonts.mx.model.forecast import DistributionForecast
+from gluonts.transform.feature import RollingMeanValueImputation
+
+from .predictor import StreamState
+from .scoring_functions import ScoringFunction, TailProbability
+
+
+class MaskingStrategy:
+    anomaly_history_size: int
+
+    def mask(
+        self,
+        masking_state: StreamState,
+        forecast: DistributionForecast,
+        values: pd.Series,
+    ) -> Tuple[StreamState, pd.Series, pd.Series]:
+        raise NotImplementedError()
+
+
+class MissingValueMaskingStrategy(MaskingStrategy):
```

Comment:
There are some changes in masking that I think should be included here also since they fix edge cases.